pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -310,13 +304,6 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
-	// validate AnomalyDetection Specs first to give error feedback if ADSpec is misconfigured
-	if err = validateConfiguredAnomalyDetectionSpec(instance.Spec.AnomalyDetection, isManagedCluster); err != nil {
```

Comment:
Do you know the source of truth is for the IntrusionDetection resource?  If it's this repo, I guess it should be OK to add that note into this PR.

Reply:
Here is the source of truth https://github.com/tigera/operator/blob/master/api/v1/intrusiondetection_types.go, you will need to run `make gen-files` as it will generate the updated CRD which also needs to be committed. It is the right action to add a deprecation notice to the CRD, this matches what was discussed in slack. The comment will also cause a deprecated note to be included in the API reference docs too when they are regenerated.

Reply:
You could add it in a separate PR but I think ideally it would be included in this PR as this PR is doing the deprecation.

Reply:
Thank you, working on this change now.